### PR TITLE
Added support for string input in tfs rest

### PR DESCRIPTION
**Pull request by bstrzele**

### CODE REVIEW

**Comment by dtrawins** [comment]
it should be rebased to develop branch
is it tested with all 4 formats: names row, unnamed row, named column and unnamed column?

**Review by dtrawins — APPROVED** [review]

File: src/test/tfs_rest_parser_binary_inputs_test.cpp [hunk 1]
```diff
@@ -66,6 +66,62 @@ TEST_F(TFSRestParserBinaryInputs, BatchSize2) {
     EXPECT_EQ(std::memcmp(parser.getProto().inputs().find("i")->second.string_val(0).c_str(), image_bytes.get(), filesize), 0);
 }
 
+TEST_F(TFSRestParserBinaryInputs, RowString) {
```
Comment (dkalinowski) [inline src/test/tfs_rest_parser_binary_inputs_test.cpp:69]: Prepare negative tests, example:
multiple precisions in the same instance
```
{"signature_name":"","inputs":{"i":["abcd", "efg", 52.1, "xyz"]}}

{"signature_name":"","inputs":{"i":[[2,3,4],[5,"abcd",7]]}}
```

File: src/test/tfs_rest_parser_binary_inputs_test.cpp [hunk 2]
```diff
@@ -66,6 +66,83 @@ TEST_F(TFSRestParserBinaryInputs, BatchSize2) {
...
+
+TEST_F(TFSRestParserBinaryInputs, ColumnStringMixedPrecision2) {
+    std::string request = R"({"signature_name":"","inputs":{"i":[[2,3,4],[5,"abcd",7]]}})";
+
+    TFSRestParser parser(prepareTensors({{"i", {-1, -1}}}, ovms ::Precision::U8));
+    ASSERT_EQ(parser.parse(request.c_str()), StatusCode::REST_COULD_NOT_PARSE_INPUT);
+}
+
+TEST_F(TFSRestParserBinaryInputs, RowString) {
+    std::string request = R"({"signature_name":"","instances":[{"i":"abcd"}]})";
```
Comment (dkalinowski) [inline src/test/tfs_rest_parser_binary_inputs_test.cpp:91]: Please add few more tests:
- json with strings but endpoint with FP32 precision
- json with strings but endpoint not 2d
- json with strings and endpoint 2d but static

File: src/test/tfs_rest_parser_binary_inputs_test.cpp [hunk 3]
```diff
@@ -98,6 +98,35 @@ TEST_F(TFSRestParserBinaryInputs, RowString) {
...
+
+TEST_F(TFSRestParserBinaryInputs, RowStringInvalidShape) {
+    std::string request = R"({"signature_name":"","instances":[{"i":"abcd"}]})";
+
+    TFSRestParser parser(prepareTensors({{"i", {-1, -1, -1}}}, ovms ::Precision::U8));
+    ASSERT_EQ(parser.parse(request.c_str()), StatusCode::OK);
+    ASSERT_EQ(parser.getProto().inputs_size(), 1);
+    ASSERT_EQ(parser.getProto().inputs().count("i"), 1);
+    ASSERT_EQ(parser.getProto().inputs().find("i")->second.string_val_size(), 1);
+    EXPECT_EQ(strcmp(parser.getProto().inputs().find("i")->second.string_val(0).c_str(), "abcd"), 0);
```
Comment (dkalinowski) [inline src/test/tfs_rest_parser_binary_inputs_test.cpp:116]: Please assert shape of such request

Comment (dkalinowski) [reply src/test/tfs_rest_parser_binary_inputs_test.cpp:116]: In all the added tests assertions for proto shape and precisions are missing @bstrzele

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**